### PR TITLE
Implementation of `sc_dot` method

### PR DESCRIPTION
**Pull request by noriakis**
Thank you for developing this very useful package.

I have implemented the `sc_dot` method for the `Seurat` and `SingleCellExperiment` objects, replicating `Seurat::DotPlot`.
https://github.com/YuLab-SMU/ProjectYulab/issues/19

I tried to make the function with the basic functionality of `Seurat::DotPlot`. Some codes are copied and adapted from the `Seurat` codebase. One `Imports` of `RColorBrewer` has been added to DESCRIPTION. R CMD CHECK passed without errors and warnings in R 4.3.1.

I would be grateful if you could review the codes for the possible integration.

### CODE REVIEW

File: R/sc-dot.R [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+	        cols <- cols[1:length(unique(splits))]
+	        names(cols) <- unique(splits)
+        }
+        d[[group.by]] <- paste(d[[group.by]], splits, sep = '_')
+        unique.splits <- unique(splits)
+        id.levels <- paste0(rep(x = id.levels, each = length(x = unique.splits)),
+        	"_", rep(x = unique(x = splits), times = length(x = id.levels)))
+    }
+    avg.exp <- d |>
+        dplyr::group_by(.data[[group.by]], features) |>
```
Comment (xiangpin) [inline R/sc-dot.R]: I think the this should using `.data[["features"]]` in here.

Comment (xiangpin) [reply R/sc-dot.R]: Or `.data$features`

File: R/sc-dot.R [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    		scaled <- scale(log1p(x))
+    		scaled <- .MinMax(scaled, min=col.min, max=col.max)
+    	} else {
+    		scaled <- log1p(x)
+    	}
+    	return(scaled)
+    }
+    
+    avg.exp <- avg.exp |>
+        dplyr::group_by(features) |>
```
Comment (xiangpin) [inline R/sc-dot.R]: `.data$features`

File: R/sc-dot.R [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+    default_mapping <- aes_string(color=color.by, size="pct.exp")
+    if (is.null(mapping)) {
+        mapping <- default_mapping
+    } else {
+        mapping <- modifyList(default_mapping, mapping)
+    }
+    if (!is.null(feature_group)) {
+        avg.exp[["feat.group"]] <- feature_group[avg.exp[["features"]]]
+    }
+	p <- ggplot(avg.exp, aes(x=features, y=.data[[group.by]])) +
```
Comment (xiangpin) [inline R/sc-dot.R]: `.data$features`

**Comment by noriakis** [comment]
Dear Dr. Xu,

Thank you very much for taking the time to review the codes. I have corrected the points you raised.

Sincerely,
Noriaki Sato

**Comment by noriakis** [comment]
Thank you very much for merging. It's a great honor to contribute to the package.

**Pull request merged**